core/core.go

```diff
@@ -303,6 +317,45 @@ func (c *Core) addToQueueIfNotAppended(block *types.Block) {
 	}
 }
 
+// SetSyncTarget sets the sync target entropy based on the prime blocks
+func (c *Core) SetSyncTarget(header *types.Header) {
+	if c.sl.subClients == nil || header.Hash() == c.sl.config.GenesisHash {
+		return
+	}
+
+	// Check if the header is in the badSyncTargets cache
+	_, ok := c.badSyncTargets.Get(header.Hash())
+	if ok {
+		return
+	}
+
+	nodeCtx := common.NodeLocation.Context()
+	// Set Sync Target for subs
+	if nodeCtx != common.ZONE_CTX {
+		if header != nil {
+			for i := range c.sl.subClients {
+				if c.sl.subClients[i] != nil {
+					c.sl.subClients[i].SetSyncTarget(context.Background(), header)
+				}
+			}
+		}
+	}
+	// In zone case
```

Comment:
This doesn't only execute in zone case. Perhaps you meant to put an `else {}` block?

Reply:
oh yeah, my bad, good catch